app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -44,7 +77,124 @@ fun MapScreen(navigationActions: NavigationActions) {
+
+                // Load the red marker image and resized it to fit the map
+                val bitmap = BitmapFactory.decodeResource(context.resources, R.drawable.red_marker)
+                val resizedBitmap = Bitmap.createScaledBitmap(bitmap, 100, 150, false)
+
+                // Get parkings in the current view
+                parkingViewModel.getParkingsInRect(bottomLeftCorner, topRightCorner)
+
+                // Create PointAnnotationOptions for each parking
+                var pointAnnotationOptions =
+                    listOfParkings.value.map { parking ->
+                      PointAnnotationOptions()
+                          .withPoint(parking.location.center)
+                          .withIconImage(resizedBitmap)
+                    }
+
+                // This is a test code snippet that can uncommented for testing purposes as per the
+                // PR description
+
+                //                val parking1 = Point.fromLngLat(6.566, 46.519)
+                //                val parking2 = Point.fromLngLat(6.500,46.500)
+                //                val parking3 = Point.fromLngLat(6.600,46.530)
+                //
+                //                val listTest = listOf(parking1,parking2,parking3)
+                //
+                //                // Create PointAnnotationOptions for each parking with test tags
+                //                val pointAnnotationOptions = listTest.mapIndexed { index, point ->
+                //
+                //                    PointAnnotationOptions()
+                //                        .withPoint(point)
+                //                        .withIconImage(resizedBitmap)
+                //                }
+
+                // Add annotations to the annotation manager and display them on the map
+                pointAnnotationOptions.forEach { annotationManager.create(it) }
+
+                // Add a camera change listener to detect zoom changes
+                val cameraChangeListener = OnCameraChangeListener {
+                  state.value = mapView.mapboxMap.cameraState.zoom
+
+                  viewportWidth = mapView.width
+                  viewportHeight = mapView.height
+
+                  centerPixel =
+                      mapView.mapboxMap.pixelForCoordinate(mapView.mapboxMap.cameraState.center)
+
+                  topRightCorner =
+                      mapView.mapboxMap.coordinateForPixel(
+                          ScreenCoordinate(
+                              centerPixel.x + viewportWidth / 2,
+                              centerPixel.y - viewportHeight / 2))
+
+                  bottomLeftCorner =
+                      mapView.mapboxMap.coordinateForPixel(
+                          ScreenCoordinate(
+                              centerPixel.x - viewportWidth / 2,
+                              centerPixel.y + viewportHeight / 2))
+
```

Comment:
The above code could be obtained from a helper function that takes as argument a mapView and returns a tuple of points (topRight, bottomLeft), to avoid duplication in the changeListener, which does the exact same thing

Reply:
This modularization will be particularly useful if in a future sprint you need to change the way you compute the corners

Reply:
I plan to do a second PR in this Sprint to improve modularization and performance.  It will aim to refactor some line of codes and handle the request to the database accordingly

Reply:
Fixed !